File: code/jee/exercises/event-list/backend/src/main/java/org/tsdes/jee/exercises/eventlist/backend/validation/CountryValidator.java
package org.tsdes.jee.exercises.eventlist.backend.validation;

import org.tsdes.jee.exercises.eventlist.backend.Countries;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

public class CountryValidator implements ConstraintValidator<Country,String> {
    @Override
    public void initialize(Country constraintAnnotation) {

    }

    @Override
    public boolean isValid(String value, ConstraintValidatorContext context) {

        return Countries.getCountries().stream()
                .anyMatch(s -> s.equalsIgnoreCase(value));
    }
}

File: spring/examples/news/src/main/kotlin/org/tsdes/spring/examples/news/constraint/CountryValidator.kt
package org.tsdes.spring.examples.news.constraint

import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

/**
 * Created by arcuri82 on 16-Jun-17.
 */
class CountryValidator : ConstraintValidator<Country, String> {

    override fun initialize(constraintAnnotation: Country) {
    }

    override fun isValid(value: String, context: ConstraintValidatorContext): Boolean {
        return CountryList.isValidCountry(value)
    }
}